betelgeuse.py

```diff
@@ -110,10 +133,163 @@ def parse_test_results(test_results):
     return Counter([test['status'] for test in test_results])
 
 
+def add_test_case(args):
+    """Task that create or update test cases and manage manage its requirement.
+
+    This task relies on ``OBJ_CACHE`` to get the collect_only and project
+    objects.
+    """
+    path, tests = args
+    collect_only = OBJ_CACHE['collect_only']
+    project = OBJ_CACHE['project']
+    requirement = None
+    for test in tests:
+        # Expect test_case_id to be path.test_name or
+        # path.ClassName.test_name.
+        test_case_id_parts = [
+            path.replace('/', '.').replace('.py', ''),
+            test.name
+        ]
+        if test.parent_class is not None:
```

Comment:
I have organized the code into blocks (separated by a blank line). At the top of this block https://github.com/SatelliteQE/betelgeuse/pull/21/files#diff-80282b6c3915f3b13ccdcac2b7c94c94R188 there is a docstring which explains the following logic.